tests/Task.test.ts

```diff
@@ -129,6 +129,34 @@ describe('parsing', () => {
         ).toStrictEqual(true);
         expect(task!.blockLink).toEqual(' ^my-precious');
     });
+
+    it('supports tag anywhere in the description and separates them correctly from signifier emojis', () => {
+        // Arrange
+        const line =
+            '- [ ] this is a task due 📅 2021-09-12 #inside_tag ⏫ #some/tags_with_underscore';
+        const path = 'this/is a path/to a/file.md';
+        const sectionStart = 1337;
+        const sectionIndex = 1209;
+        const precedingHeader = 'Eloquent Section';
+
+        // Act
+        const task = Task.fromLine({
+            line,
+            path,
+            sectionStart,
+            sectionIndex,
+            precedingHeader,
+        });
+
+        // Assert
+        expect(task).not.toBeNull();
+        expect(task!.description).toEqual(
+            'this is a task due #inside_tag #some/tags_with_underscore',
+        );
+        expect(task!.dueDate).not.toBeNull();
+        expect(task!.dueDate!.isSame(moment('2021-09-12', 'YYYY-MM-DD')));
+        expect(task!.priority == Priority.High);
+    });
```

Comment:
This is a comment on `Tasks.ts` parsing, but noting here. as I discovered it by adding a kind of stress test:

I added this new test, and it failed. I intentionally added all the fields in the reverse order that they were parsed, to make it harder work for the current implementation to parse.

```typescript
    it('supports parsing large number of values', () => {
        // Arrange
        const line =
            '- [ ] Wobble ⏫  #tag1 ✅ 2022-07-02 #tag2  📅 2022-07-02 #tag3 ⏳ 2022-07-02 #tag4 🛫 2022-07-02 #tag5  🔁 every day  #tag6 #tag7 #tag8 #tag9 #tag10';

        // Act
        const task = fromLine({
            line,
        });

        // Assert
        expect(task).not.toBeNull();
        expect(task!.description).toEqual(
            'Wobble #tag1 #tag2 #tag3 #tag4 #tag5 #tag6 #tag7 #tag8 #tag9 #tag10',
        );
        expect(task!.dueDate!.isSame(moment('022-07-02', 'YYYY-MM-DD')));
        expect(task!.doneDate!.isSame(moment('022-07-02', 'YYYY-MM-DD')));
        expect(task!.startDate!.isSame(moment('022-07-02', 'YYYY-MM-DD')));
        expect(task!.scheduledDate!.isSame(moment('022-07-02', 'YYYY-MM-DD')));
        expect(task!.priority == Priority.High);
        expect(task!.tags).toStrictEqual([
            '#tag1',
            '#tag2',
            '#tag3',
            '#tag4',
            '#tag5',
            '#tag6',
            '#tag7',
            '#tag8',
            '#tag9',
            '#tag10',
        ]);
    })
```

The first failure is:

```
Error: expect(received).toEqual(expected) // deep equality

Expected: "Wobble #tag1 #tag2 #tag3 #tag4 #tag5 #tag6 #tag7 #tag8 #tag9 #tag10"
Received: "Wobble ⏫  #tag1 ✅ 2022-07-02 #tag2  📅 2022-07-02 #tag3 #tag4 #tag5 #tag6 #tag7 #tag8 #tag9 #tag10"
```

I think the problem is this code:

```text
        // Add a "max runs" failsafe to never end in an endless loop:
        const maxRuns = 7;
```

It used to be possible to reason about the maximum possible number of elements in the task (assuming that the user never duplicated any of the supported values like due date, etc)

But now that tags can be mixed in, if there is a hard-coded limit like this, it will always be possible to use one more tag than the code was designed for, and blow out the parsing loop limit.

I think that _may_ mean that the existing parsing code needs a bit more of a re-think???

Reply:
> I think that may mean that the existing parsing code needs a bit more of a re-think???

Or maybe just increase the value of `maxRuns` quite a bit???

Reply:
Awesome catch. One solution is that when a tag is parsed and removed, it would not eat a run out of `maxRuns`, because tags are unlimited and task component have just a few possibilities.
This will work, but it makes me think about an issue that probably exited in the code since forever: nothing currently prevents the user from entering multiple due dates, multiple recurrence rules etc. They will override each other and `maxRuns` may not suffice for such a user case regardless of the tags change.
But since another goal of `maxRuns` is to limit the runtime of task parsing (e.g. so a user with huge lines that are accidentally parsed as tasks will not bog down all of Obsidian), perhaps the wisest thing is also the simplest, which is to just increase `maxRuns` to 20 without other changes.
What do you think?

Reply:
> One solution is that when a tag is parsed and removed, it would not eat a run out of maxRuns, because tags are unlimited and task component have just a few possibilities.

I do think that's a good idea.

I also wondered about whether some users might put several tags in, in a row, and maybe the number of passes through the passing might be reduced by putting the current `find a tag`-type block in a `while loop`. It would make the code more complicated, but that could perhaps be improved later by breaking it out in to a function.

On reflection, though, I think that `while` loop idea can be left for another PR though, if it turns out there really is a problem...

> This will work, but it makes me think about an issue that probably exited in the code since forever: nothing currently prevents the user from entering multiple due dates, multiple recurrence rules etc. They will override each other and maxRuns may not suffice for such a user case regardless of the tags change.

Indeed. And they will override each other in a non-obvious way by picking the earlier value rather than the later one.

But I'm not worried about that at this point, as:

1. it was a pre-existing behaviour,
2. it's obvious to the user that there is a duplicate,
3. and I would prefer to keep this PR focused one one change...

> But since another goal of maxRuns is to limit the runtime of task parsing (e.g. so a user with huge lines that are accidentally parsed as tasks will not bog down all of Obsidian), 

That's an interesting observation. I think you may be right. I had interpreted it purely as stopping infinite loops.

> perhaps the wisest thing is also the simplest, which is to just increase maxRuns to 20 without other changes.

> What do you think?

I think on reflection either stopping tags eating the run limit, or upping `maxRuns` to 20 or 30 would be good. They are both nice and simple.

Any further changes can be considered for later on, based on more use and on feedback from users.

Reply:
I went ahead with the "simpler" approach of just changing `maxRuns` to 20, and added your test to verify it now passes.
The idea of having tags not consume from `maxRuns` may be more bullet-proof in parsing long lists of tags, but it also introduces the concern that huge lines that are not actually intended to be tasks (e.g. in some imported data that users hold in the vault), and have long lists that include `#...` tokens, will bring Obsidian to a halt. With a strict all-included max value we can be more sure this plugin isn't causing performance concerns.